src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,30 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (obj instanceof Herbivore && obj.health > 0 && !obj.hidden) {
```

Comment:
The check for `obj.health > 0` is unnecessary here. The health of the animal should be checked after it has been decreased by the bite, and if it becomes less than or equal to 0, the animal should be removed from the `Animal.alive` array.